store/todos.js

```diff
@@ -133,26 +153,54 @@ export const mutations = {
   SET_IS_SEARCHING: (state, status) => {
     state.isSearching = status
   },
+  SET_IS_TODO_LOADING: (state, status) => {
+    state.isTodoLoading = status
+  },
+  SET_IS_TODO_LIST_LOADING: (state, status) => {
+    state.isTodoListLoading = status
+  },
+  SET_LOADING_ID: (state, id) => {
+    state.loadingId = id
```

Comment:
What is loadingId used here for?

Reply:
`loadingId` is used to show loader for specific todo item.

Reply:
Design issues resolved. Thanks

Reply:
Is it necessary to maintain a loading ID for specific todo? Can we make it component specific?

Reply:
Then, we have to maintain a loading state in every item of the list, right ?

Reply:
Each `<TodoItem />` can have a loader that will be conditionally rendered based on some logic. I think in this way, we don't have to maintain a global loading ID.

Reply:
Resolved. Thanks